cattle/plugins/docker/network/setup.py

```diff
@@ -13,7 +13,16 @@ def __init__(self):
         pass
 
     def before_start(self, instance, host, config, start_config):
-        pass
+        mac_address = None
+        deviceNumber = None
+        for nic in instance.nics:
+            if deviceNumber is None:
+                mac_address = nic.macAddress
+                deviceNumber = nic.deviceNumber
+            elif deviceNumber < nic.deviceNumber:
```

Comment:
You want >, not <, right?

Also, you need a test that exercises this logic. It can be in the same test function, but add a test where you have more than one nic and assert that you properly get the nic with the lowest deviceNumber.